### PR TITLE
Remove IEntityDto restrictions from CrudApplicationService

### PR DESCRIPTION
**Pull request by ahmednfwela**
### Description

This PR removes the need to use IEntityDto when using Abp's `AppService`s, specifically this line:

```csharp
where TEntityDto : IEntityDto<TKey>
```

There is no need to enforce any rules regarding Dtos, especially since they can be provided by an external library (e.g. protobuf messages)

affected classes:
- ReadOnlyAppService
- CrudAppService

### Checklist

- [ ] I fully tested it as developer / designer and created unit / integration tests
- [ ] I documented it (or no need to document or I will create a separate documentation issue)

### CODE REVIEW

**Comment by ahmednfwela** [comment]
Test failure is unrelated
```
2023-11-20T16:16:13.6792727Z Test run for /home/runner/work/abp/abp/framework/test/Volo.Abp.Settings.Tests/bin/Debug/net8.0/Volo.Abp.Settings.Tests.dll (.NETCoreApp,Version=v8.0)
2023-11-20T16:16:13.6832850Z Starting test execution, please wait...
2023-11-20T16:16:13.7357310Z A total of 1 test files matched the specified pattern.
2023-11-20T16:16:13.8586146Z Microsoft (R) Test Execution Command Line Tool Version 17.8.0 (x64)
2023-11-20T16:16:13.8587854Z Copyright (c) Microsoft Corporation.  All rights reserved.
2023-11-20T16:16:13.8687242Z 
2023-11-20T16:16:14.0543721Z Starting test execution, please wait...
2023-11-20T16:16:14.1361258Z A total of 1 test files matched the specified pattern.
2023-11-20T16:16:18.5517007Z [xUnit.net 00:00:01.44]     Volo.Abp.Security.Claims.AbpDynamicClaimsPrincipalContributorBase_Tests.AddDynamicClaimsAsync [FAIL]
2023-11-20T16:16:19.7108792Z   Failed Volo.Abp.Security.Claims.AbpDynamicClaimsPrincipalContributorBase_Tests.AddDynamicClaimsAsync [680 ms]
2023-11-20T16:16:19.7110614Z   Error Message:
2023-11-20T16:16:19.7111621Z    Shouldly.ShouldAssertException : claimsPrincipal.Identities.First().Claims
2023-11-20T16:16:19.7112870Z     should contain an element satisfying the condition
2023-11-20T16:16:19.7114205Z ((c.Type == .SurName) && (c.Value == "test-family_name"))
2023-11-20T16:16:19.7115072Z     but does not
2023-11-20T16:16:19.7115625Z   Stack Trace:
2023-11-20T16:16:19.7118167Z      at Volo.Abp.Security.Claims.AbpDynamicClaimsPrincipalContributorBase_Tests.AddDynamicClaimsAsync() in /home/runner/work/abp/abp/framework/test/Volo.Abp.Security.Tests/Volo/Abp/Security/Claims/AbpDynamicClaimsPrincipalContributorBase_Tests.cs:line 65
2023-11-20T16:16:19.7121307Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2023-11-20T16:16:19.7122789Z    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2023-11-20T16:16:19.7125527Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task, ConfigureAwaitOptions options)
2023-11-20T16:16:19.7127512Z --- End of stack trace from previous location ---
2023-11-20T16:16:19.7128830Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2023-11-20T16:16:19.7130315Z    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2023-11-20T16:16:19.7132357Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task, ConfigureAwaitOptions options)
2023-11-20T16:16:19.7134546Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2023-11-20T16:16:19.7136044Z    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2023-11-20T16:16:19.7138139Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task, ConfigureAwaitOptions options)
2023-11-20T16:16:19.7329621Z 
2023-11-20T16:16:19.7336371Z Failed!  - Failed:     1, Passed:     9, Skipped:     0, Total:    10, Duration: 216 ms - Volo.Abp.Security.Tests.dll (net8.0)
```

File: framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/ReadOnlyAppService.cs [hunk 1]
```diff
@@ -22,8 +21,7 @@ protected ReadOnlyAppService(IReadOnlyRepository<TEntity, TKey> repository)
 
 public abstract class ReadOnlyAppService<TEntity, TEntityDto, TKey, TGetListInput>
     : ReadOnlyAppService<TEntity, TEntityDto, TEntityDto, TKey, TGetListInput>
-    where TEntity : class, IEntity<TKey>
-    where TEntityDto : IEntityDto<TKey>
+    where TEntity : class, IEntity<TKey>
```
Comment (ahmednfwela) [inline framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/ReadOnlyAppService.cs]:
```suggestion
    where TEntity : class, IEntity<TKey>
```

File: framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/ReadOnlyAppService.cs [hunk 2]
```diff
@@ -34,9 +32,7 @@ protected ReadOnlyAppService(IReadOnlyRepository<TEntity, TKey> repository)
 
 public abstract class ReadOnlyAppService<TEntity, TGetOutputDto, TGetListOutputDto, TKey, TGetListInput>
     : AbstractKeyReadOnlyAppService<TEntity, TGetOutputDto, TGetListOutputDto, TKey, TGetListInput>
-    where TEntity : class, IEntity<TKey>
-    where TGetOutputDto : IEntityDto<TKey>
-    where TGetListOutputDto : IEntityDto<TKey>
+    where TEntity : class, IEntity<TKey>
```
Comment (ahmednfwela) [inline framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/ReadOnlyAppService.cs]:
```suggestion
    where TEntity : class, IEntity<TKey>
```

**Comment by maliming** [comment]
hi

We can't use the `Id` without `IEntityDto<TKey>`.

**Comment by ahmednfwela** [comment]
@maliming I am sorry I don't understand why? automapper uses property name not interface

**Comment by maliming** [comment]
hi @ahmednfwela 

You're right, I will check it again.

**Comment by maliming** [comment]
Can you merge latest dev to your branch?

```
ERROR: Permission to Bdaya-Dev/abp.git denied to maliming.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by ahmednfwela** [comment]
@maliming Done

**Comment by ahmednfwela** [comment]
CI fail is unrelated
```
2023-12-19T13:19:13.1747667Z Test run for /home/runner/work/abp/abp/framework/test/Volo.Abp.Settings.Tests/bin/Debug/net8.0/Volo.Abp.Settings.Tests.dll (.NETCoreApp,Version=v8.0)
2023-12-19T13:19:13.3485393Z Microsoft (R) Test Execution Command Line Tool Version 17.8.0 (x64)
2023-12-19T13:19:13.3491239Z Copyright (c) Microsoft Corporation.  All rights reserved.
2023-12-19T13:19:13.3589498Z 
2023-12-19T13:19:13.5974514Z Starting test execution, please wait...
2023-12-19T13:19:13.6806525Z A total of 1 test files matched the specified pattern.
2023-12-19T13:19:16.9376173Z [xUnit.net 00:00:01.15]     Volo.Abp.Security.Claims.AbpDynamicClaimsPrincipalContributorBase_Tests.AddDynamicClaimsAsync [FAIL]
2023-12-19T13:19:17.1117293Z   Failed Volo.Abp.Security.Claims.AbpDynamicClaimsPrincipalContributorBase_Tests.AddDynamicClaimsAsync [683 ms]
2023-12-19T13:19:17.1244514Z   Error Message:
2023-12-19T13:19:17.1245349Z    Shouldly.ShouldAssertException : claimsPrincipal.Identities.First().Claims
2023-12-19T13:19:17.1246408Z     should contain an element satisfying the condition
2023-12-19T13:19:17.1247618Z ((c.Type == .UserName) && (c.Value == "test-preferred_username"))
2023-12-19T13:19:17.1248310Z     but does not
2023-12-19T13:19:17.1408734Z   Stack Trace:
2023-12-19T13:19:17.1411752Z      at Volo.Abp.Security.Claims.AbpDynamicClaimsPrincipalContributorBase_Tests.AddDynamicClaimsAsync() in /home/runner/work/abp/abp/framework/test/Volo.Abp.Security.Tests/Volo/Abp/Security/Claims/AbpDynamicClaimsPrincipalContributorBase_Tests.cs:line 64
2023-12-19T13:19:17.1414984Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2023-12-19T13:19:17.1416934Z    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2023-12-19T13:19:17.1419077Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task, ConfigureAwaitOptions options)
2023-12-19T13:19:17.1421625Z --- End of stack trace from previous location ---
2023-12-19T13:19:17.1422889Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2023-12-19T13:19:17.1424750Z    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2023-12-19T13:19:17.1426848Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task, ConfigureAwaitOptions options)
2023-12-19T13:19:17.1429553Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2023-12-19T13:19:17.1431105Z    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2023-12-19T13:19:17.1433589Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task, ConfigureAwaitOptions options)
2023-12-19T13:19:18.4251431Z 
2023-12-19T13:19:18.4314617Z Failed!  - Failed:     1, Passed:     9, Skipped:     0, Total:    10, Duration: 501 ms - Volo.Abp.Security.Tests.dll (net8.0)
```

**Review by maliming — APPROVED** [review]

**Comment by maliming** [comment]
Thanks @ahmednfwela 👍

**Pull request merged**